firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -454,6 +460,7 @@ private PersistedInstallationEntry registerFidWithServer(PersistedInstallationEn
 
     switch (response.getResponseCode()) {
       case OK:
+        cachedFid = response.getFid();
```

Comment:
I recommend to remove this and only set cachedFid when you successfully read from SharedPrefs.
You solution is probably better because 1 less read of SharedPrefs, but potentially more error cases.